llama_index/indices/base.py

```diff
@@ -256,3 +257,23 @@ def as_query_engine(self, **kwargs: Any) -> BaseQueryEngine:
         if "service_context" not in kwargs:
             kwargs["service_context"] = self._service_context
         return RetrieverQueryEngine.from_args(**kwargs)
+
+    def as_chat_engine(
+        self, chat_mode: ChatMode = ChatMode.CONDENSE_QUESTION, **kwargs: Any
+    ) -> BaseChatEngine:
```

Comment:
i'd probably add a line `chat_mode = ChatMode(chat_mode)` if chat_mode is passed in as a string (which should be allowed for easier UX, and is also fine typing wise since ChatMode inherits from both str and enum)

Reply:
> btw @Disiok heads up, the paul graham data link in condense question ipynb is broken

Good catch fixed.

Reply:
> i'd probably add a line `chat_mode = ChatMode(chat_mode)` if chat_mode is passed in as a string (which should be allowed for easier UX, and is also fine typing wise since ChatMode inherits from both str and enum)

why is that necessary? for type checking?